pkgs/development/php-packages/box/default.nix

```diff
@@ -29,5 +29,6 @@ mkDerivation {
     license = licenses.mit;
     homepage = "https://github.com/box-project/box";
     maintainers = with maintainers; [ jtojnar ] ++ teams.php.members;
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

Comment:
PHAR is closer to a tarball containing plain-text PHP scripts (+ bootstrap script, possibly compressed) so it is not really native code. Since the scripts would be executed by PHP interpreter/JIT compiler, `binaryBytecode` sounds closer to me. But it does not really fit either – the code is still in plaintext form inside the archive.

cc @risicle Maybe we need new provenance like `sourceInBinaryArtefact`.

Reply:
@risicle WDYT ?

Reply:
Yup, I've been feeling there's a space for something like this. Various things we see include:

 - Various places where we use pre-built webapp assets in otherwise-source packages because it's painful to assemble ourselves. I was tempted at one point to call these `minifiedSource` but I'm not sure if that's broad enough.
 - Packages like `sqlite` where we use source that has been pre-processed to a single bundled source file.
 - If we want to take it to extremes, many packages we download as "release" source archives (as opposed to raw copies of the repo) have had their autotools run and may have had some of their repo's directories removed, rearranged or augmented.

I've wondered if some of these could be classified as something like `preprocessedSource`?

The question I guess we've got to ask ourselves is "why do we care?". And I think once we're talking about different _levels_ of build-from-source, of all the concerns we have about source provenance there are really just two remaining:

 - Patchability. It's usually _possible_ with pre-processed source, but not much fun and will require manual patch mangling at the very least.
 - Inspectability. While something we consider to be a "source" is usually readable to some degree, it's not always easy to verify that what you're working with truly corresponds to the original unadulterated source.

I certainly don't want to hold up a real code-changing PR with our navel-gazing though, so what *I* would do in cases like this for now is *not* mark them as anything non-source but leave a comment making them easy to find if/when we do decide to introduce a more fitting classification.

If you're concerned about grabbing a tarball that includes native binaries which you don't use, I sometimes make a show of going in and deleting them in a `prePatch` (`postUnpack`?) hook and then explicitly marking the source provenance as `fromSource` to show you _have_ thought about this.

Reply:
Thanks for your reply.

I'll remove those from the PR then.